Feat/navigation component

nanana3679 opened the pull request:
### 구현 목록

- Navigation Rail https://m3.material.io/components/navigation-rail/overview
- Navigation Drawer https://m3.material.io/components/navigation-drawer/overview
- Navigation Bar https://m3.material.io/components/navigation-bar/overview
- Top App Bar https://m3.material.io/components/top-app-bar/overview
    - 좌측 메뉴 아이콘 대신 Navigation Bar 사용
- 최상위 layout.tsx에 반응형 레이아웃 적용
    - 미디어 쿼리
    
### M3 컴포넌트 래핑
  
 - Icon, IconButton https://material-web.dev/components/icon-button/
   - .svg 대신 아이콘 폰트 `<link />` 사용 
 - Menu https://material-web.dev/components/menu/

nanana3679 committed:
feat: add m3 Icon/IconButton with examples

nanana3679 committed:
feat: add Icon variations

nanana3679 committed:
feat: implement navigation bar component

nanana3679 committed:
feat: implement navigation rail component

nanana3679 committed:
feat: implement navigation drawer component

nanana3679 committed:
feat: add M3 menu component with examples

nanana3679 committed:
feat: implement top app bar component

nanana3679 committed:
feat: add responsive navigation layout

nanana3679 committed:
fix: unify navigation icon color

nanana3679 committed:
fix: hide navigation drawer scrim for extra-large window size

nanana3679 committed:
Merge branch 'frontend-main' into feat/navigation-component

nanana3679 committed:
fix: resolve layout and navigation nesting structure

- Move navigation and i18n providers to locale layout
- Separate layout styles into dedicated SCSS file
- Remove duplicate HTML structure

nanana3679 commented:
@jms0214 layout끼리 겹쳐서 구조를 바꿨는데 확인해주세요
![image](https://github.com/user-attachments/assets/d7958fd3-34df-452f-918f-231d074b8fa4)
![image](https://github.com/user-attachments/assets/0512ef14-23ef-460c-b14b-b7b2098b804d)

jms0214 commented on frontend/src/app/page.tsx:
app 아래에 page랑 layout 둬도 에러 안나는지?

jms0214 commented on frontend/src/app/[locale]/providers/i18n-provider.tsx:
파일 이름을 컴포넌트 이름과 동일하게 I18nProvider.tsx로 하면 어떨까?
Next.js에서 시스템 파일 (layout.tsx, page.tsx, error.tsx등)은 소문자로 시작하는 컨벤션,
React 컴포넌트 파일은 PascalCase를 따르면 좋을 것 같음.

jms0214 commented on frontend/src/app/layout.tsx:
```diff
@@ -0,0 +1,43 @@
+  return routing.locales.map((locale) => ({ locale }));
+}
+
+export default async function LocaleLayout({ children }: { children: React.ReactNode }) {
```
최상위 layout 의 function 이름은 RootLayout이 좋을 것 같음.

jms0214 commented on frontend/src/app/page.tsx:
```diff
@@ -0,0 +1,34 @@
+
+import { IconButtonExample } from '@/components/IconButton/IconButtonExample';
+
+export default function Home() {
```
저번에 이야기했던 대로 Home 대신 HomePage로

jms0214 reviewed (commented):
리뷰 확인 부탁

nanana3679 committed:
feat: replace state layer to M3 Ripple component

nanana3679 committed:
chore: improve layout naming convention

- rename outer layout to RootLayout
- rename i18n-provider.tsx to I18nProvider.tsx
- delete outer page.tsx

nanana3679 commented:
@jms0214 의견 반영 후 머지합니다
